### PR TITLE
CAL-249 Update the video streaming code to configure chunks in megabytes

### PR DESCRIPTION
**Pull request by glenhein**
#### What does this PR do?
Update the video streaming code to configure chunks in megabytes instead of bytes.

#### Who is reviewing it (please choose AT LEAST two reviewers that need to approve the PR before it can get merged; if a component team is listed, at least one of its members needs to approve)?
@bdeining 
@jrnorth 

#### Choose 2 committers to review/merge the PR (please choose ONLY two committers from below, delete the rest).
@jlcsmith
@lessarderic
#### How should this be tested?
1) Build and install.
2) With both the FMV Admin Plugin and the video stream configuration page, create a stream monitor. Note that the default values are now in megabytes. Re-open the config and confirm that the re-loaded values for the chunk size are correct. 
3) Stream video and confirm that the chunk sizes are as expected.

#### What are the relevant tickets?

[CAL-249](https://codice.atlassian.net/browse/CAL-249)
#### Checklist:
- [ ] Documentation Updated
	- [ ] Change Log Updated
- [X] Update / Add Unit Tests
- [X] Update / Add Integration Tests

### CODE REVIEW

File: catalog/video/video-mpegts-stream/src/test/java/org/codice/alliance/video/stream/mpegts/rollover/ByteCountRolloverConditionTest.java [hunk 1]
```diff
@@ -26,6 +26,9 @@
 
 public class ByteCountRolloverConditionTest {
 
+    /**
+     * megabytes
+     */
     private static final long THRESHOLD = 100;
```
Comment (bdeining) [inline catalog/video/video-mpegts-stream/src/test/java/org/codice/alliance/video/stream/mpegts/rollover/ByteCountRolloverConditionTest.java]: nit : you could rename this to MEGABYTE_THRESHOLD and remove the comment

Comment (glenhein) [inline catalog/video/video-mpegts-stream/src/test/java/org/codice/alliance/video/stream/mpegts/rollover/ByteCountRolloverConditionTest.java]: ok

File: catalog/video/video-mpegts-stream/src/main/resources/OSGI-INF/metatype/metatype.xml [hunk 2]
```diff
@@ -33,9 +33,9 @@
                 type="String" default="" />
 
         <AD
-                description="Maximum file size (bytes) before rollover. Must be >=1."
+                description="Maximum file size (megabytes) before rollover. Must be >=1."
                 name="Max File Size" id="byteCountRolloverCondition" required="false"
```
Comment (jrnorth) [inline catalog/video/video-mpegts-stream/src/main/resources/OSGI-INF/metatype/metatype.xml]: `megabyteCountRolloverCondition`?

File: catalog/video/video-mpegts-stream/src/main/java/org/codice/alliance/video/stream/mpegts/rollover/ByteCountRolloverCondition.java [hunk 3]
```diff
@@ -23,26 +24,43 @@
...
-    private long byteCountThreshold;
+    private static final long MEGABYTE_TO_BYTE_FACTOR = 1000000;
 
-    public ByteCountRolloverCondition(long byteCountThreshold) {
-        this.byteCountThreshold = byteCountThreshold;
+    public static final long MIN_VALUE = 1;
+
+    public static final long MAX_VALUE = Long.MAX_VALUE / MEGABYTE_TO_BYTE_FACTOR;
+
+    private long megabyteCountThreshold;
```
Comment (jlcsmith) [inline catalog/video/video-mpegts-stream/src/main/java/org/codice/alliance/video/stream/mpegts/rollover/ByteCountRolloverCondition.java]: I'm wondering if we should make the MB threshold a double.  Thoughts?

File: catalog/video/video-mpegts-stream/src/main/resources/OSGI-INF/metatype/metatype.xml [hunk 4]
```diff
@@ -33,9 +33,9 @@
                 type="String" default="" />
 
         <AD
-                description="Maximum file size (bytes) before rollover. Must be >=1."
+                description="Maximum file size (megabytes) before rollover. Must be >=1."
```
Comment (jlcsmith) [inline catalog/video/video-mpegts-stream/src/main/resources/OSGI-INF/metatype/metatype.xml]: If we don't make the max file size a double then we should add a note here that whole numbers only are allowed.

File: catalog/video/video-mpegts-stream/src/main/java/org/codice/alliance/video/stream/mpegts/rollover/ByteCountRolloverCondition.java [hunk 5]
```diff
@@ -23,26 +24,43 @@
  */
 public class ByteCountRolloverCondition implements RolloverCondition {
```
Comment (bdeining) [inline catalog/video/video-mpegts-stream/src/main/java/org/codice/alliance/video/stream/mpegts/rollover/ByteCountRolloverCondition.java]: Should the class be renamed?

File: catalog/video/video-mpegts-stream/src/main/resources/OSGI-INF/metatype/metatype.xml (hunk 4 above)
Comment (bdeining) [inline catalog/video/video-mpegts-stream/src/main/resources/OSGI-INF/metatype/metatype.xml]: Agreed.  It's worth noting that the video admin plugin drops the decimal values and uses the whole number

File: catalog/video/video-mpegts-stream/src/main/java/org/codice/alliance/video/stream/mpegts/rollover/ByteCountRolloverCondition.java (hunk 3 above)
Comment (bdeining) [inline catalog/video/video-mpegts-stream/src/main/java/org/codice/alliance/video/stream/mpegts/rollover/ByteCountRolloverCondition.java]: My opinion is that making this a double would introduce some unnecessary rounding complexity.  Is there a use case for specifying 22.2 MB?

**Comment by bdeining** [comment]
Hero Build successful.  Verified video chunks are being set and created appropriately according to the configuration 👍  LGTM after last comments are addressed.

File: catalog/video/video-admin-plugin/src/main/webapp/js/model/StreamMonitor.js [hunk 6]
```diff
@@ -80,7 +80,7 @@ define(['backbone',
                     $.each(configurations, function(index, value) {
                         var url = value.monitoredAddress;
                         var maxDuration = value.elapsedTimeRolloverCondition / (60 * 1000);
-                        var maxSize = value.byteCountRolloverCondition / (1000 * 1000);
+                        var maxSize = value.byteCountRolloverCondition;
```
Comment (lessarderic) [inline catalog/video/video-admin-plugin/src/main/webapp/js/model/StreamMonitor.js]: ✏️ If these values are in MB, should we name them `maxMegabyteSize` and `megabyteCountRolloverCondition`?

File: catalog/video/video-admin-plugin/src/main/webapp/js/view/StreamMonitor.view.js [hunk 7]
```diff
@@ -186,7 +186,7 @@ define([
                 var url = this.$(".feedUrl").val().split("/").join("!/");
                 var networkInterface = this.isMulticastUrl(this.$(".feedUrl").val()) ? this.$(".networkInterface").val() : null;
                 var elapsedTimeRolloverCondition = parseInt(this.$(".feedMaxDuration").val()) * 60 * 1000;
-                var byteCountRolloverCondition = parseInt(this.$(".feedMaxClipSize").val()) * 1000 * 1000;
+                var byteCountRolloverCondition = parseInt(this.$(".feedMaxClipSize").val());
```
Comment (lessarderic) [inline catalog/video/video-admin-plugin/src/main/webapp/js/view/StreamMonitor.view.js]: ✏️ Same question as above.

File: catalog/video/video-mpegts-stream/src/test/java/org/codice/alliance/video/stream/mpegts/rollover/ByteCountRolloverConditionTest.java [hunk 8]
```diff
@@ -59,16 +59,16 @@ public void testCurrentEqualToThreshold() {
...
 
         assertThat(condition.isRolloverReady(packetBuffer), is(true));
 
     }
 
     @Test
     public void testSetter() {
-        condition.setByteCountThreshold(THRESHOLD - 1);
-        assertThat(condition.getByteCountThreshold(), is(THRESHOLD - 1));
+        condition.setMegabyteCountThreshold(MEGABYTE_THRESHOLD - 1);
```
Comment (lessarderic) [inline catalog/video/video-mpegts-stream/src/test/java/org/codice/alliance/video/stream/mpegts/rollover/ByteCountRolloverConditionTest.java]: ❗️ We probably should test the setter and constructor validation edge cases (below min and over max). We could also just keep the setter and remove the constructor and not have to test the latter.

File: catalog/video/video-mpegts-stream/src/main/resources/OSGI-INF/metatype/metatype.xml (hunk 2 above)
Comment (glenhein) [inline catalog/video/video-mpegts-stream/src/main/resources/OSGI-INF/metatype/metatype.xml]: sure, I can change the name

File: catalog/video/video-mpegts-stream/src/main/resources/OSGI-INF/metatype/metatype.xml (hunk 4 above)
Comment (glenhein) [inline catalog/video/video-mpegts-stream/src/main/resources/OSGI-INF/metatype/metatype.xml]: Chnaged: description="Maximum file size (megabytes) before rollover. Must be >=1 and integer."

File: catalog/video/video-mpegts-stream/src/main/java/org/codice/alliance/video/stream/mpegts/rollover/ByteCountRolloverCondition.java (hunk 5 above)
Comment (glenhein) [inline catalog/video/video-mpegts-stream/src/main/java/org/codice/alliance/video/stream/mpegts/rollover/ByteCountRolloverCondition.java]: ok

File: catalog/video/video-admin-plugin/src/main/webapp/js/model/StreamMonitor.js (hunk 6 above)
Comment (glenhein) [inline catalog/video/video-admin-plugin/src/main/webapp/js/model/StreamMonitor.js]: ok

File: catalog/video/video-admin-plugin/src/main/webapp/js/view/StreamMonitor.view.js (hunk 7 above)
Comment (glenhein) [inline catalog/video/video-admin-plugin/src/main/webapp/js/view/StreamMonitor.view.js]: changing the names

File: catalog/video/video-mpegts-stream/src/test/java/org/codice/alliance/video/stream/mpegts/rollover/ByteCountRolloverConditionTest.java (hunk 8 above)
Comment (glenhein) [inline catalog/video/video-mpegts-stream/src/test/java/org/codice/alliance/video/stream/mpegts/rollover/ByteCountRolloverConditionTest.java]: i'll add a few more tests for the edge cases

File: catalog/video/video-mpegts-stream/src/main/resources/OSGI-INF/metatype/metatype.xml (hunk 2 above)
Comment (glenhein) [inline catalog/video/video-mpegts-stream/src/main/resources/OSGI-INF/metatype/metatype.xml]: renamed

File: catalog/video/video-mpegts-stream/src/main/resources/OSGI-INF/metatype/metatype.xml (hunk 4 above)
Comment (glenhein) [inline catalog/video/video-mpegts-stream/src/main/resources/OSGI-INF/metatype/metatype.xml]: added to description saying the value must be an integer

File: catalog/video/video-mpegts-stream/src/main/java/org/codice/alliance/video/stream/mpegts/UdpStreamMonitor.java [hunk 9]
```diff
@@ -523,15 +523,15 @@ public Integer getByteCountRolloverCondition() {
     /**
      * @param count must be non-null and positive
      */
-    public void setByteCountRolloverCondition(Integer count) {
+    public void setMegabyteCountRolloverCondition(Integer count) {
         notNull(count, "count must be non-null");
 
         inclusiveBetween(BYTE_COUNT_MIN,
```
Comment (jrnorth) [inline catalog/video/video-mpegts-stream/src/main/java/org/codice/alliance/video/stream/mpegts/UdpStreamMonitor.java]: Will these limits need to be changed?

File: catalog/video/video-mpegts-stream/src/test/java/org/codice/alliance/video/stream/mpegts/UdpStreamMonitorTest.java [hunk 10]
```diff
@@ -69,20 +69,20 @@ public void testSetStartImmediately() {
...
     @Test(expected = NullPointerException.class)
     public void testSetByteCountRolloverConditionNullArg() {
-        udpStreamMonitor.setByteCountRolloverCondition(null);
+        udpStreamMonitor.setMegabyteCountRolloverCondition(null);
     }
 
     @Test(expected = IllegalArgumentException.class)
     public void testSetByteCountRolloverConditionBelowRangeArg() {
-        udpStreamMonitor.setByteCountRolloverCondition(UdpStreamMonitor.BYTE_COUNT_MIN - 10);
+        udpStreamMonitor.setMegabyteCountRolloverCondition(UdpStreamMonitor.BYTE_COUNT_MIN - 10);
```
Comment (jrnorth) [inline catalog/video/video-mpegts-stream/src/test/java/org/codice/alliance/video/stream/mpegts/UdpStreamMonitorTest.java]: Should that constant be renamed to `MEGABYTE_COUNT_MIN`?

File: catalog/video/video-mpegts-stream/src/main/java/org/codice/alliance/video/stream/mpegts/UdpStreamMonitor.java (hunk 9 above)
Comment (glenhein) [inline catalog/video/video-mpegts-stream/src/main/java/org/codice/alliance/video/stream/mpegts/UdpStreamMonitor.java]: I changed the limits to use the min and max values defined in MegabyteRolloverCondition

File: catalog/video/video-mpegts-stream/src/test/java/org/codice/alliance/video/stream/mpegts/UdpStreamMonitorTest.java (hunk 10 above)
Comment (glenhein) [inline catalog/video/video-mpegts-stream/src/test/java/org/codice/alliance/video/stream/mpegts/UdpStreamMonitorTest.java]: renamed, and renamed the test method
